tests/test_subscriptions.py

```diff
@@ -191,6 +195,8 @@ def test_find(self):
         self.assertEqual(self.endpoint, Endpoint.from_subscription(json_response['subscriptions'][0]))
         self.assertNotIn('hmac_secret_key', json_response['subscriptions'][0])
         self.assertEqual(num_additions, len(json_response['subscriptions']))
+        for _uuid in uuids:
+            self.addCleanup(self._cleanup_subscription, _uuid)
```

Comment:
This needs to put at the beginning of the function. I suggest around line 183. Alternatively you could put in `_put_subscription` so that every time a subscriptions is created a cleanup subscription is added to the test teardown.